### ISSUE TITLE
Analysis: Use NREL instead of Esri for Charging Station layer?

### ISSUE THREAD

**Issue by NicoledeGreef** [bcgov/smk-clean-bc#80]
> [Open in MURAL](https://app.mural.co/t/datasystemsandservices8142/m/datasystemsandservices8142/1674081183025?wid=0-1675834839284) ![image](https://user-images.githubusercontent.com/2573142/30230004-a70b171c-94ba-11e7-95f6-53290185d9b7.png)
> CleanBC JIRA to GH Integration - CITZ – BCDS – Data Systems and Services

**Comment by michaelpnelson** [comment]
# Outcome
PlugShare’s API doesn’t appear to give driving directions to a charging location, but it does have lots to offer in finding charging locations and providing data about them. 

**An API key is required.** A key can be requested at  [API Request Form](https://developer.plugshare.com/access) . However this page currently says that only commercial licenses (i.e. not private or non-commercial) are available at this time. We would expect Clean BC to get and maintain an API key.

Key entities in their API (https://developer.plugshare.com/docs ) are Location, Station, and Outlet. A Location contains one or more Stations, and a Station contains one or more Outlets. 

API for Location (https://developer.plugshare.com/docs/#location ) seems to be most useful to us. It has methods to:

* Get Locations nearby (via lat/long or address)
* Get Locations in a region via lat/long spans
* Search Locations using a query, which can include the name of a city or state/province and several other attributes
* Fetch Locations along a route, via a route (polyline of lat/longs) plus a distance in meters from the route

**Comment by michaelpnelson** [comment]
Copied from our JIRA issue:

Instead of using the Esri-managed layer, we could instead use the source of the Esri data from the National Renewable Energy Laboratory (NREL).

Their data can be viewed in this station locator: [Alternative Fuels Data Center: Alternative Fueling Station Locator](https://afdc.energy.gov/stations/#/analyze?country=CA&region=CA-BC) 

NREL has an API for finding charging stations (
https://developer.nrel.gov/docs/transportation/alt-fuel-stations-v1/all/ ) that can return several formats including GeoJSON, but it requires using an API key (API Key Usage  | NREL: Developer Network )
This worked  though ... https://developer.nrel.gov/api/alt-fuel-stations/v1.geojson?api_key=DEMO_KEY&access=public&fuel_type=ELEC&country=CA&state=BC

Proof of concept using captured JSON: [Comparing main...nrel_layer · bcgov/smk-clean-bc](https://github.com/bcgov/smk-clean-bc/compare/main...nrel_layer) 

I’ve also confirmed I can use the API directly with the demo key, though we would want to get an API key of our own.

It looks like the points are identical.

# Outcome
Using NREL data directly is a feasible alternative to using Esri’s hosted copy of the data. An API key is required for NREL’s API; if Clean BC chooses to use their data, we would ask them to acquire and maintain the key.

**Comment by michaelpnelson** [comment]
Hi @bethbyoung ! I wanted to mention something before our meeting next week. We did some analysis and found we can use NREL data directly for EV stations in Canada instead of Esri's layer, which we noticed isn't always updated accurately. If this sounds good to you, we suggest you get an API key for us to use in the app. There's more context and links in the previous comment on this issue, and the signup page is https://developer.nrel.gov/signup/.
